testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs

```diff
@@ -642,45 +643,69 @@ impl SerializedTx {
 
 #[derive(Debug, Clone, Deserialize)]
 #[serde(rename_all = "camelCase")]
-pub struct UTXO {
+pub struct ParsedUTXO {
     txid: String,
     vout: u32,
     script_pub_key: String,
-    amount: serde_json::Number,
+    amount: Box<RawValue>,
     confirmations: u32,
     spendable: bool,
     solvable: bool,
     desc: Option<String>,
     safe: bool,
 }
 
-impl UTXO {
+pub struct UTXO {
+    txid: Sha256dHash,
+    vout: u32,
+    script_pub_key: Script,
+    amount: u64,
+}
+
+impl ParsedUTXO {
 
-    pub fn get_txid(&self) -> Sha256dHash {
-        let mut txid = hex_bytes(&self.txid).expect("Invalid byte sequence");
-        txid.reverse();
-        Sha256dHash::from(&txid[..])
+    pub fn get_txid(&self) -> Option<Sha256dHash> {
+        match hex_bytes(&self.txid) {
+            Ok(ref mut txid) => {
+                txid.reverse();
+                Some(Sha256dHash::from(&txid[..]))
+            }
+            Err(err) => {
+                warn!("Unable to get txid from UTXO {}", err);
+                None
+            }
+        }
     }
 
-    pub fn get_sat_amount(&self) -> u64 {
-        UTXO::serialized_btc_to_sat(&self.amount.to_string())
+    pub fn get_sat_amount(&self) -> Option<u64> {
+        ParsedUTXO::serialized_btc_to_sat(self.amount.get())
     }
 
-    pub fn serialized_btc_to_sat(amount: &str) -> u64 {
+    pub fn serialized_btc_to_sat(amount: &str) -> Option<u64> {
         let comps: Vec<&str> = amount.split(".").collect();
         match comps[..] {
             [lhs, rhs] => {
-                let base: u64 = 10;
-                let btc_to_sat = base.pow(8);
-                let btc = lhs.parse::<u64>().expect("Invalid amount");
-                let mut amount = btc * btc_to_sat;
-                assert!(rhs.len() <= 8, "Unexpected amount of decimals");
-                let frac_part = rhs.parse::<u64>().expect("Invalid amount");
-                let sat = frac_part * base.pow(8 - rhs.len() as u32);
-                amount += sat;
-                amount
+                match (lhs.parse::<u64>(), rhs.parse::<u64>()) {
+                    (Ok(btc), Ok(frac_part)) => {
+                        if rhs.len() > 8 {
```

Comment:
This check probably belongs before the `rhs.parse::<u64>()` match statement.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1538/commits/c064f76dc1782fd5cb8cc280e5f037619d75e20e